sentry/src/routes/campaign.rs

```diff
@@ -292,6 +293,53 @@ pub async fn campaign_list<A: Adapter>(
     Ok(success_response(serde_json::to_string(&list_response)?))
 }
 
+/// Can only be called by creator
+/// to close a campaign, just set it's budget to what it's spent so far (so that remaining == 0)
+/// newBudget = totalSpent, i.e. newBudget = oldBudget - remaining
+pub async fn close_campaign<A: Adapter>(
+    req: Request<Body>,
+    app: &Application<A>,
+) -> Result<Response<Body>, ResponseError> {
+    let auth = req
+        .extensions()
+        .get::<Auth>()
+        .expect("Auth should be present");
+
+    let mut campaign = req
+        .extensions()
+        .get::<Campaign>()
+        .expect("We must have a campaign in extensions")
+        .to_owned();
+
+    if auth.uid.to_address() != campaign.creator {
+        Err(ResponseError::Forbidden(
+            "Request not sent by campaign creator".to_string(),
+        ))
+    } else {
+        let remaining = app
+            .campaign_remaining
+            .get_remaining_opt(campaign.id)
+            .await?
+            .map(|remaining| UnifiedNum::from(max(0, remaining).unsigned_abs()))
+            .ok_or_else(|| {
+                ResponseError::BadRequest("No remaining entry for campaign".to_string())
+            })?;
```

Comment:
- [ ] Please use https://redis.io/commands/getset
This will ensure that redis will be updated first and any subsequent events will fail with "No budget left".